be/src/exec/pipeline/adaptive/event.h

```diff
@@ -0,0 +1,63 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <atomic>
+#include <memory>
+#include <vector>
+
+#include "exec/pipeline/adaptive/adaptive_fwd.h"
+#include "exec/pipeline/pipeline_fwd.h"
+#include "runtime/runtime_state.h"
+
+namespace starrocks::pipeline {
```

Comment:
this feature is very excellent, however, can we re-design like this;

Event e0;
Event e1;

Event e2 = Event.dependsAnyOn(e0, e1);

Event e2 = Event.dependsAllOn(e0, e1);

e2 is ready only when e0 and e1 is finished.